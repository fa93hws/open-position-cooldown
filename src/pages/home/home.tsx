import * as React from 'react';
import { Box } from '@material-ui/core';
import { withTheme, WithTheme } from '@material-ui/core/styles';

import { IntentionService } from '@services/intention/intention';
import { LocalStorageService } from '@services/local-storage/local-storage';
import { createCards } from './cards/cards';
import { AddButton } from './add-button/add-button';
import { createHomeDrawer } from './drawer/drawer';

type HomePageProps = WithTheme & {
  IconButton: React.ComponentType;
  Cards: React.ComponentType;
  Sheet: React.ComponentType;
  onMount(): void;
};
export const HomePage = React.memo(
  withTheme((props: HomePageProps) => {
    React.useEffect(() => {
      props.onMount();
    }, []);
    return (
      <Box height="100%" position="relative">
        <props.Cards />
        <Box position="absolute" bottom={16} right={16}>
          <props.IconButton />
        </Box>
        <props.Sheet />
      </Box>
    );
  }),
);

export function createHomePage() {
  const localStorageService = new LocalStorageService();
  const intentionService = new IntentionService(localStorageService);
  const [Cards, cardsStore] = createCards(intentionService);

  const afterSubmit = () => cardsStore.fetchIntentions();
  const { Component, showDrawer } = createHomeDrawer(
    intentionService,
    afterSubmit,
  );

  const AddButtonImpl = () => <AddButton onClick={showDrawer} />;

  const onMount = () => cardsStore.fetchIntentions();

  return () => (
    <HomePage
      IconButton={AddButtonImpl}
      Sheet={Component}
      Cards={Cards}
      onMount={onMount}
    />
  );
}
