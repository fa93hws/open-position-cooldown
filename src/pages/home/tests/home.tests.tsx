import * as React from 'react';
import { render, cleanup } from '@testing-library/react';

import { createHomePage, HomePage } from '../home';

describe('HomePage', () => {
  beforeEach(cleanup);

  it('renders this', () => {
    const IconButton = () => (
      <button>
        <i>icon!</i>
      </button>
    );
    const Sheet = () => <div>sheet!</div>;
    const Cards = () => <div>Cards!</div>;
    expect(
      <HomePage
        IconButton={IconButton}
        Sheet={Sheet}
        Cards={Cards}
        onMount={jest.fn()}
      />,
    ).toMatchRenderedSnapshot();
  });

  it('is able to be mounted', () => {
    const Component = createHomePage();
    expect(() => render(<Component />)).not.toThrow();
  });
});
