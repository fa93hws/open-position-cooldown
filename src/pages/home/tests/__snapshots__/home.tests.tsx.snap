// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders this 1`] = `
<div
  class="MuiBox-root MuiBox-root-1"
>
  <div>
    Cards!
  </div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <button>
      <i>
        icon!
      </i>
    </button>
  </div>
  <div>
    sheet!
  </div>
</div>
`;
